### PR TITLE
Gui: Handle various property types better in VarSet dialog

### LINKED ISSUE #15156: Gui: Various property types in the VarSet dialog have unexpected results.
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

The dialog to add properties to VarSets (command `Std_VarSet`) presents the user with a list of properties. Other than the standard dialog to add properties, this dialog allows users to directly add values. Some of the properties in the list do not allow directly adding a value or should not be user facing.

I propose to remove from the list the properties below because these are more internal properties and they cause an exception for the user:
- App::Property
- App::PropertyComplexGeoData
- ExpressionContainer
- ExpressionEngine
- Geometry
- IntegerSet
- LinkBase
- LinkListBase
- Lists
- XLinkContainer
- Sketcher::PropertyConstraintList

The following properties should not have an editor in this dialog (so they should be edited in the property view):
- Direction
- Placementlist
- Position

The reason for this is that these properties require a "compound" editor for multiple value input.

The following property type does not work properly:
- Font

### Full version info

```shell
OS: Arch Linux
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38025 (Git)
Build type: DEBUG
Branch: main
Hash: 04cc890fd0685bc4143de1a8000e92dce0580d28
Python 3.12.4, Qt 6.7.2, Coin 4.0.2, Vtk , OCC 7.7.2
Locale: English/United States (en_US)
Installed mods: 
  * OpenTheme 2024.5.16
  * ondsel-lens 2024.7.2.01
```

### Subproject(s) affected?

Core

### Anything else?

Reported [here](https://github.com/Ondsel-Development/FreeCAD/issues/58#issuecomment-2204805080)

I've tested all property types.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by pieterhijma**
This PR makes a better distinction between property types that should not be supported because they are more for internal use, for example as base classes, such as `App::Property` and `App::PropertyLinkBase`.

It also loads font data for `App::PropertyFont`

Closes #15156

Note that there are other issues related as well. For example, https://github.com/FreeCAD/FreeCAD/issues/15159 has a similar problem with properties that result in exceptions.

### CODE REVIEW

**Comment by furgo16** [comment]
Related PR for the DlgAddProperty counterpart: https://github.com/FreeCAD/FreeCAD/pull/15191

**Comment by pieterhijma** [comment]
The PR has updated to align with (and build on) https://github.com/FreeCAD/FreeCAD/pull/15191, so please merge that one first, so this PR can be rebased on that one. (This PR contains the commit from #15191). The two dialogs use the same mechanism to filter out property types that lead to exceptions. 

As such, @maxwxyz, I would recommend to mark this as 1.0 because users may simply get unnecessary exceptions which is solved by this PR and also #15191. I would also mark that one as 1.0 for the same reason.

File: src/Gui/DlgAddPropertyVarSet.cpp [hunk 1]
```diff
@@ -101,19 +101,30 @@ void DlgAddPropertyVarSet::initializeGroup()
...
+        return strcmp(a.getName(), b.getName()) < 0;
+    });
+}
+
 void DlgAddPropertyVarSet::initializeTypes()
 {
     auto paramGroup = App::GetApplication().GetParameterGroupByPath(
             "User parameter:BaseApp/Preferences/PropertyView");
-    auto lastType = Base::Type::fromName(
+   auto lastType = Base::Type::fromName(
```
Comment (chennes) [inline src/Gui/DlgAddPropertyVarSet.cpp:120]:
```suggestion
    auto lastType = Base::Type::fromName(
```

**Review by chennes — APPROVED** [review]
Just one incredibly minor comment (you lost a space somehow), otherwise this looks good to me. That said, since it touches BaseClass.h (and therefore effectively the entire codebase, I'd prefer to have @wwmayer give it a look and merge if it is OK.

**Comment by wwmayer** [comment]
Most of the changes come from my PR https://github.com/FreeCAD/FreeCAD/pull/15191/files

File: src/Gui/DlgAddPropertyVarSet.cpp (hunk 1 above)
Comment (pieterhijma) [reply src/Gui/DlgAddPropertyVarSet.cpp]: Thanks, my difftool didn't show this for some reason, just fixed it. The changes of `Baseclass.h` are from Werner's commit in PR #15191 (that I recommend to merge first and that I base my changes on), so effectively this PR only makes changes to the VarSet dialog.

**Pull request merged**